aiohttp/web_protocol.py

```diff
@@ -483,17 +478,21 @@ async def start(self) -> None:
                 task = self._loop.create_task(self._handle_request(request, start))
                 try:
                     resp, reset = await task
+                except HTTPException as exc:
```

Comment:
Please move this `except` block and `if getattr(resp, "__http_exception__", False):` into `_handle_request method`.